include/seastar/core/deleter.hh

```diff
@@ -113,7 +114,7 @@ private:
 
 /// \cond internal
 struct deleter::impl {
-    unsigned refs = 1;
+    std::atomic_uint refs = 1;
```

Comment:
```suggestion
    std::atomic<unsigned int> refs = 1;
```

Reply:
Seastar (and usual) style is to use atomic<> template, not these typedefs which I think are just there for C compatibility.

Reply:
Switching to your recommendation now.